tests/test_expval.py

```diff
@@ -169,6 +171,8 @@ def test_sprod(self, diff_method, qubit_device):
         """Test the `SProd` class with lightning qubit."""
 
         dev = qubit_device(wires=2)
+        if diff_method == "adjoint" and dev.short_name == "lightning.qubit2":
+            return
```

Comment:
Why do we have this (and others), if tests are skipped for the new API.

Reply:
Not sure. I'll remove it. Nice catch!

Reply:
I think these changes predate the addition of adjoint support and could be updated now.